File: source/constructs/lambda/api/api-sm-param.py
import logging
import boto3

logger = logging.getLogger()
logger.setLevel(logging.INFO)

client = boto3.client('secretsmanager')

def lambda_handler(event, context):
    result = []
    
    response = client.list_secrets(SortOrder='asc')
    for secret in response.get('SecretList'):
        result.append({'name' : secret.get('Name'), 'description' : secret.get('Description')})
    
    next_token = response.get('NextToken')
    
    while next_token is not None:
        response = client.list_secrets(SortOrder = 'asc', NextToken = next_token)
        for secret in response.get('SecretList'):
            result.append({'name' : secret.get('Name'), 'description' : secret.get('Description')})
        
        next_token = response.get('NextToken')

    logger.info("Get %d secrets" % len(result))

    return result

File: source/constructs/lambda/api/api-sm-param.ts
import * as AWS from "aws-sdk";
import { Context } from "aws-lambda";
import { pprint } from '../common';

type Parameter = {
    name?: string,
    description?: string
}

const handler = async function (event: any, context?: Context) {

    pprint('EVENT', event)
    pprint('CONTEXT', context)

    const sm = new AWS.SecretsManager({apiVersion: '2017-10-17'});
    let result: Array<Parameter> = [];

    const params = {
        MaxResults: 100,
    };

    await sm.listSecrets(params, function (err, data) {
        if (err) console.log(err, err.stack); // an error occurred
        else {
            console.log(data);
            if (data.SecretList) {
                for (const s of data.SecretList) {
                    result.push({
                        name: s.Name,
                        description: s.Description,
                    })
                }
            }
        }
    }).promise();

    return result
}

export { handler }